docs/source/sql.rst

```diff
@@ -65,6 +65,227 @@ route when they are available.::
     sqlalchemy.Table <- CSV
 
 
+Primary and Foreign Key Relationships
+-------------------------------------
+
+.. versionadded:: 0.3.4
+
+.. warning::
+
+   Primary and foreign key relationship handling is an experimental feature and
+   is subject to change.
+
+Odo has experimental support for creating and discovering relational database
+tables with primary keys and foreign key relationships.
+
+Creating a new resource with a primary key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We create a new ``sqlalchemy.Table`` object with the resource function,
+specifying the primary key in the ``dshape`` argument
+
+   .. code-block:: python
+
+      >>> from odo import resource
+      >>> dshape = 'var * {id: !int64, name: string}'
+      >>> products = resource('sqlite:///db.db::products', dshape=dshape)
+      >>> products.c.id.primary_key
+      True
```

Comment:
Primary-ness might be part of the name rather than the dtype?  E.g.

    {!id: int64, ...}